CONTRIBUTING.md

```diff
@@ -550,6 +550,252 @@ cat generated/watcher/watcher.log  # See k_ids field for each core in the last d
   ensure that your commit message follows our required format as outlined above
   in this section. Failure to do so is a violation of our standards.
 
+### Git commit runbook
+
+The tt-metal repository has a single main branch, with pull requests being
+merged in via rebase or squash merge.
+
+For a pull request, you will need to deal with a local branch and an origin
+branch.
+
+The local branch should be on your machine, and you should interact with it
+through the command line or Visual Studio. The following uses the command line.
+
+The origin branch is on Github and you should use the Github UI to interact
+with the branch.
+
+#### Creating the initial Pull Request (PR)
+
+You need to have an issue. Either someone creates one and assigns it to you, or
+you need to create an issue. For example issue 123.
+
+#### Creating a branch on a machine
+
+Include the user, the issue number, and optionally a description of the change.
+/ and - are used as separators between user and issue number. And - and _
+between issue number and description. E.g.
+
+```
+git checkout -b user-123
+git checkout -b user/123
+git checkout -b user-123_rename_method_x
+git checkout -b user/123-add-x-unit-test
+```
+
+#### Saving your changes
+
+Edit the files that you want, making sure relevant unit tests pass. Then add
+them in. E.g.
+
+```
+git add abc.py
+git add "*.py"
+```
+
+Please avoid using `git add -A`, which is fairly error prone.
+
+You can restore files if you need to get the original. E.g.
+
+```
+git restore abc.py
+git restore '*'
+git restore --staged abc.py # if the file was already added
+```
+
+Once you are satisfied that everything works, create a commit. The message
+needs to specify the issue number, with a pound sign and a colon, and the
+description should describe the change itself, and cannot be generic and refer
+to something else, such as saying that you are addressing reviewer feedback.
```

Comment:
You can reference the other part of the document that has the commit message standards in full.

Reply:
added